### DISCUSSION TITLE
Repeatable annotations

### DISCUSSION THREAD

**Discussion by ssuryaprabhakar** [ozlerhakan/poiji#288] [General]
I've a parent class (Address) which is then inherited by several children (Billing-Address, Shipping-Address etc.)
These are separate columns in the spreadsheet. How do I annotate the parent and map to multiple columns in the sheet?

**Comment by ozlerhakan** [comment]
Hi @ssuryaprabhakar , Does this https://github.com/ozlerhakan/poiji#super-class-inheritance help your need?

**Comment by ssuryaprabhakar** [comment]
In your example, the "name" column is repeated in the sheet with different column headers (mapping it to my example, BillingAddr1, ShippingAddr1)

@ExcelCell(0)
protected String name;

**Reply by ozlerhakan** [reply]
Could you please explain your case with examples?

**Reply by ssuryaprabhakar** [reply]
This is the parent class. Address can be home-address, business-address, and alternate-address. All of these are separate columns in the Excel. How do I annotate the fields with multiple values.

for ex. 

@ExcelCellName(value = "homeAddr1", mandatoryCell = true)
@ExcelCellName(value = "businessAddr1", mandatoryCell = true)
@ExcelCellName(value = "AlternateAddr1", mandatoryCell = false)
private String address1;



public class Address {
	private String company;
	private String contact;
	private String address1;
	private String address2;
	private String address3;
	private String city;
	private String state;
	private String postalCode;
	private String country;
	private String phone;
	private String fax;
	private String email;

**Reply by ozlerhakan** [reply]
Poiji does not offer this feature. You need to add three separate fields for three individual columns and once you get the values of addresses use depending on the priority order: 

```
get address() {
  if address1 != null
      return address1
  
  if address2 != null
      return address2
  
  if address3 != null
      return address3

  return null

```
